### PR TITLE
Add quote taxes

### PR DESCRIPTION
**Pull request by lexerom**
## Summary

Add taxes to Quote, rearrange tax files. So, invoice and quote will have the same tax object (with some readonly properties), CreditMemo will have tax with all writable properties.
[Preview](https://preview.redoc.ly/all-rebilly/add-quote-taxes/)

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by noon-dawg — DISMISSED** [review]
Looks good on the docs side

**Review by tamasvigh — DISMISSED** [review]

File: openapi/components/schemas/Taxes/TaxTypes/rebilly-taxjar.yaml [hunk 1]
```diff
@@ -12,11 +12,11 @@ properties:
     type: integer
     x-type: Money
     x-sortable: true
-    description: Total amount of all invoice taxes.
+    description: Total amount of all taxes.
```
Comment (adamaltman) [inline openapi/components/schemas/Taxes/TaxTypes/rebilly-taxjar.yaml:15]:
```suggestion
    description: Total amount of taxes.
```

File: openapi/components/schemas/UpcomingInvoice.yaml [hunk 2]
```diff
@@ -70,7 +70,9 @@ properties:
   shipping:
     $ref: ./Shipping.yaml
   tax:
-    $ref: ./Invoices/InvoiceTax.yaml
+    description: Invoice taxes.
+    allOf:
+      - $ref: ./Taxes/Taxes.yaml
```
Comment (adamaltman) [inline openapi/components/schemas/UpcomingInvoice.yaml:75]: I would use the description from the Taxes.yaml object everywhere throughout and not try to override the description with `allOf`.

**Review by adamaltman — CHANGES_REQUESTED** [review]

File: openapi/components/schemas/Taxes/TaxTypes/rebilly-avalara.yaml [hunk 3]
```diff
@@ -12,11 +12,11 @@ properties:
...
     x-sortable: true
-    description: Total amount of all invoice taxes.
+    description: Total amount of all taxes.
     readOnly: true
   items:
     type: array
-    description: Taxes that are applied to this invoice and calculated based on the `taxCategoryId` of the related product.
+    description: Taxes applied and calculated based on the `taxCategoryId` of the related product.
     items:
       allOf:
```
[the remark this answers is not in the document]
Comment (lexerom) [reply openapi/components/schemas/Taxes/TaxTypes/rebilly-avalara.yaml]: Fixed.

**Review by noon-dawg — DISMISSED** [review]

File: openapi/components/schemas/Taxes/CreditMemoTaxItem.yaml [hunk 4]
```diff
@@ -1,33 +1,33 @@
 allOf:
   - properties:
       rate:
-        readOnly: true
+        readOnly: false
```
Comment (adamaltman) [inline openapi/components/schemas/Taxes/CreditMemoTaxItem.yaml:4]: `allOf` does not override. This is not valid.

**Review by adamaltman — CHANGES_REQUESTED** [review]
You need two schemas.

**Review by adamaltman — APPROVED** [review]

**Review by tamasvigh — APPROVED** [review]

**Pull request merged**